src/commands.rs

```diff
@@ -15,24 +33,33 @@ pub struct Args<'m> {
 }
 
 pub(crate) struct Commands {
-    state_machine: StateMachine,
+    state_machine: StateMachine<Arc<Command>>,
     client: HttpClient,
-    menu: HashMap<&'static str, &'static str>,
+    menu: Option<HashMap<&'static str, (&'static str, GuardFn)>>,
 }
 
 impl Commands {
     pub(crate) fn new() -> Self {
         Self {
             state_machine: StateMachine::new(),
             client: HttpClient::new(),
-            menu: HashMap::new(),
+            menu: Some(HashMap::new()),
```

Comment:
Why an option?

Reply:
The ownership semantics [here][1] and [here][2] are kinda difficult to deal with when borrowing the menu HashMap.  Using an Option here, I can move the menu out of its allocation as a member of `Commands` and replace it with a None. This way I get an owned instance of the menu in main.  

[1]: https://github.com/rust-lang/discord-mods-bot/blob/089759bc5ff43af3879e0cd2633f025b7d639212/src/commands.rs#L172

[2]: https://github.com/rust-lang/discord-mods-bot/blob/089759bc5ff43af3879e0cd2633f025b7d639212/src/main.rs#L148-L150